components/script/dom/globalscope.rs

```diff
@@ -642,7 +659,7 @@ fn stream_handle_incoming(stream: &ReadableStream, bytes: Fallible<Vec<u8>>, can
 
 /// Callback used to close streams as part of FileListener.
 fn stream_handle_eof(stream: &ReadableStream) {
-    stream.close_native();
+    stream.close();
```

Comment:
pretty sure this needs to be `close_native`, and this could be noticeable because `close` unconditionally closes the stream, while `close_native` does a "request close" if there are still chunks in the queue. This doesn't show up in tests because the stream is emptied by the reader before this is called, I guess. 

(note we should also rename `close_native` as per the other comment)